tap_typeform/streams.py

```diff
@@ -1,406 +1,316 @@
+                                                    form_id, max_bookmark, state, start_date)
+            page_count = response.get('page_count', 0)
+            if records:
+                params['before'] = records[-1].get('token')
+
+        write_bookmarks(self.tap_stream_id, selected_stream_ids, form_id, max_bookmark, state)
+        singer.write_state(state)
 
+class FullTableStream(Stream):
+    endpoint = 'forms/{}'
+
+    replication_method = 'FULL_TABLE'
+
+    def sync_obj(self, client, state, catalogs, form,
+                    start_date, selected_stream_ids):
+        full_url = client.build_url(self.endpoint).format(form)
+        stream_catalog = get_schema(catalogs, self.tap_stream_id)
+        response = client.request(full_url, params=self.params)
+
+        for record in response[self.data_key]:
+            self.add_fields_at_1st_level(record,{"form_id": form})
+
+        write_records(stream_catalog, self.tap_stream_id, response[self.data_key])
+
+class Forms(IncrementalStream):
+    tap_stream_id = 'forms'
+    key_properties = ['id']
+    endpoint='forms'
+    replication_method = 'INCREMENTAL'
+    replication_keys = ['last_updated_at']
+    data_key = 'items'
+    params = {'page_size': FORMS_PAGE_SIZE}
+
+    def get_forms(self, client):
+        full_url = client.build_url(self.endpoint)
+        page = 1
+        paginate = True
+        params = {**self.params}
+        while paginate:
+            params['page'] = page
+            response = client.request(full_url, params=params)
+            page_count = response.get('page_count')
+            paginate = page_count > page
+            page += 1
+            yield response.get(self.data_key)
+
+    def sync_obj(self, client, state, catalogs,
+                    start_date, selected_stream_ids):
+        bookmark = state.get('bookmarks',{}).get(self.tap_stream_id,{}).get(self.replication_keys[0], start_date)
+        max_bookmark = bookmark
+
+        for records in self.get_forms(client):
+            max_bookmark = self.write_records(records, catalogs, selected_stream_ids,
+                        None, max_bookmark, state, start_date)
+            write_bookmarks(self.tap_stream_id, selected_stream_ids, None, max_bookmark, state)
 
-def sync_landings(atx, form_id):
-    # similar to `answers` stream use now_parsed to fallback on and to bookmark
-    now_parsed = pendulum.now(pytz.utc)
-    default_start_date = atx.config['start_date'] or now_parsed.isoformat()
-    bookmark = get_bookmark_value(atx.state, 'landings', form_id, default=default_start_date)
-    bookmark_dt = pendulum.parse(bookmark)
-    max_dt = bookmark_dt
+        singer.write_state(state)
 
-    response = get_landings(atx, form_id, int(bookmark_dt.timestamp()))
+class Questions(FullTableStream):
+    tap_stream_id = 'questions'
+    key_properties = ['form_id', 'question_id']
+    endpoint = 'forms/{}'
+    params = {
+                'since': '',
```

Comment:
Can you please remove this 'since' param? As it isn't used for the FULL_TABLE stream.

Reply:
Removed the param